dd-trace-core/src/main/java/datadog/trace/core/util/SystemProvider.java

```diff
@@ -0,0 +1,18 @@
+package datadog.trace.core.util;
+
+/**
+ * A pluggable system provider used by {@linkplain SystemAccess}. {@linkplain SystemAccess} may not
+ * use JMX classes (even via transitive dependencies) due to potential race in j.u.l initialization.
+ * Therefore it uses an abstract {@linkplain SystemProvider} type to hold the actual implementation
+ * which may be switched between the {@linkplain SystemProvider#NONE} and {@linkplain
+ * JmxSystemProvider} on-the-fly once JMX is safe to use.
+ */
+public interface SystemProvider {
```

Comment:
This provider is providing for `SystemAccess` - hence `SystemAccessProvider` would convey the intention better, IMO.

```suggestion
public interface SystemAccessProvider {
```